Control/public/Model.js

```diff
@@ -297,14 +297,25 @@ export default class Model extends Observable {
   }
 
   /**
-   * After pages load check if the user enabled notifications
+   * Request notification permission
    */
-  checkBrowserNotificationPermissions() {
+  requestBrowserNotificationPermissions() {
     if (Notification.permission !== 'granted') {
       Notification.requestPermission();
     }
   }
 
+  /**
+   * Defines policy when user can trigger request for enabling notifications
+   * This is used to display/hide enable button
+   * @returns {bool} True when notifications are not enabled and HTTPS is used
+   */
+  checkBrowserNotificationPermissions() {
+    return (Notification.permission === 'denied' ||
+      Notification.permission === 'default') &&
+      window.location.protocol === "https:";
```

Comment:
As per the API, the `Notification` is only provided by the browser in `HTTPS` protocol so there is no need to check on it.